shared/util/preconditions.go

```diff
@@ -12,63 +12,103 @@ import (
+// Defaults run all validations
+func NewValidationCheckList() *ValidationCheckList {
+	return &ValidationCheckList{
+		MinDockerComposeVersion:          true,
+		DockerInstalled:                  true,
+		DockerEngineRunning:              true,
+		AwsInstalled:                     true,
+		TerraformInstalled:               true,
+		AwsSessionManagerPluginInstalled: true,
 	}
+}
 
-	v, err := exec.CommandContext(ctx, "docker", "compose", "version", "--short").Output()
-	if err != nil {
-		errs = multierror.Append(errs, errors.Wrap(err, "could not determine docker compose version"))
-	}
+func ValidateEnvironment(ctx context.Context, checklist *ValidationCheckList) error {
 
-	version := strings.TrimSpace(string(v))
-	dockerComposeVersion, err := semver.NewVersion(version)
-	if err != nil {
-		return errors.Wrapf(err, `invalid docker compose version. docker compose >= V2 required but "%s" was detected, please follow https://docs.docker.com/compose/cli-command/`, version)
+	if checklist == nil {
+		checklist = NewValidationCheckList()
 	}
-	valid, reasons := dockerComposeMinVersion.Validate(dockerComposeVersion)
-	if !valid {
-		errs = multierror.Append(
-			errs,
-			errors.Errorf("docker compose >= V2 required but %s was detected, please follow https://docs.docker.com/compose/cli-command/", version),
-		)
-		for _, reason := range reasons {
-			errs = multierror.Append(errs, reason)
+
+	var errs *multierror.Error
+
+	if checklist.DockerInstalled {
+		_, err := exec.LookPath("docker")
+		if err != nil {
+			errs = multierror.Append(errs, errors.Wrap(err, "could not find docker in path"))
 		}
 	}
 
-	_, err = exec.LookPath("aws")
-	if err != nil {
-		errs = multierror.Append(errs, errors.Wrap(err, "could not find aws cli in path, run 'brew install awscli' or follow https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html"))
-	}
+	if checklist.MinDockerComposeVersion {
+
+		dockerComposeMinVersion, err := semver.NewConstraint(">= v2")
+		if err != nil {
+			return errors.Wrap(err, "could not establish docker compose version")
+		}
+
+		v, err := exec.CommandContext(ctx, "docker", "compose", "version", "--short").Output()
+		if err != nil {
+			errs = multierror.Append(errs, errors.Wrap(err, "could not determine docker compose version"))
+		}
+
+		version := strings.TrimSpace(string(v))
+		dockerComposeVersion, err := semver.NewVersion(version)
+		if err != nil {
+			return errors.Wrapf(err, `invalid docker compose version. docker compose >= V2 required but "%s" was detected, please follow https://docs.docker.com/compose/cli-command/`, version)
+		}
+		valid, reasons := dockerComposeMinVersion.Validate(dockerComposeVersion)
+		if !valid {
+			errs = multierror.Append(
+				errs,
+				errors.Errorf("docker compose >= V2 required but %s was detected, please follow https://docs.docker.com/compose/cli-command/", version),
+			)
+			for _, reason := range reasons {
+				errs = multierror.Append(errs, reason)
+			}
+		}
```

Comment:
I would recommend breaking up these as individual functions and using them in the respective commands validation checks. See link above.

Reply:
Agreed. Ok - I'll work on it. Should be done with it today.